input/script_utils.go

```diff
@@ -1323,6 +1323,54 @@ func CommitScriptToRemoteConfirmed(key *btcec.PublicKey) ([]byte, error) {
 	return builder.Script()
 }
 
+// TaprootCommitScriptToRemote constructs a taproot witness program for the
+// output on the commitment transaction for the remote party. For the top level
+// key spend, we'll use the combined funding key (musig2.KeyAgg(k1, k2)), as a
+// sort of practical NUMs point (the local party would never sign for this). We
```

Comment:
I think the reason is that there needs to be a sequence check (the script is `<remotepubkey> OP_CHECKSIG
OP_CHECKSEQUENCEVERIFY`) so there needs to be an internal pubkey. It doesn't make much sense to make it the revocation pubkey except to save a few bytes on-chain in case of a breach. I think the rationale to not use a NUMS point was that "how do we know this point is truly NUMS". But if it makes recovery harder, then maybe it should be changed to NUMS

Reply:
Correct that we _can_ just use a NUMs point here. In early review of the spec proposal, people felt it was simpler to just use the multi-sig keys here, since there's an incentive to not sign for this path. 

Good point here about recovery though, I hadn't thought about that yet (also not implemented yet). Why would it make it impossible to scan though? Assuming that you have your SCB, then you have both multi-sig keys, then you can do the combine, and you also know that the script itself is just the remote key w/ no CSV, so you can always reproduce it, no?

I see what you mean about scanning though, so the idea is that if it's the NUMs point, then they don't even need the SCB, and all they need is their seed as we can walk the HD keychain to find things on chain?

Reply:
> I see what you mean about scanning though, so the idea is that if it's the NUMs point, then they don't even need the SCB, and all they need is their seed as we can walk the HD keychain to find things on chain?

Exactly. It's currently possible with `static_to_remote` and `anchor_output` channels to recover funds _without_ SCB if the remote party force closed, just by generating all possible addresses and looking them up in an address index or through an API. With a NUMS point that would be possible again.
Not sure how strong of an argument this is, since you're always supposed to keep the SCB. But I believe it's one of the more often used commands of `chantools` currently, so people seem to lose their SCB quite often.

Reply:
IMO it's the right move, but I got some push back from the initial spec reviewers since they weren't as familiar with the concept. We _do_ have [this library](https://github.com/lightninglabs/lightning-node-connect/tree/master/mailbox/numsgen) (which I used to generate the point that was in the original draft), which they can all check/replicate to ensure it's actually a NUMs point. I can raise this at the upcoming spec meeting.

Reply:
Yeah forgot about the revealing of the delay script, I guess a simple public NUMs can do the job then.

Reply:
@yyforyongyu re the NUMs key: the remote party should have a different script key each time, so the'll have a distinct top-level public key (you basically tweak the NUMs key with the combined funding key).

Reply:
PR to restore NUMS key here: https://github.com/Roasbeef/lightning-rfc/pull/6

(will also use it to make the other main change: put the breach path in the script tree for anchors)